internal/users/idgenerator.go

```diff
@@ -289,7 +164,8 @@ func getIDCandidate(minID, maxID uint32, usedIDs []uint32) (id uint32, uniqueIDs
 
 	// Fallback: try IDs from the minimum ID up to the preferred ID.
 	for id := minID; id < preferredID && id <= maxID; id++ {
-		if pos, found := slices.BinarySearch(usedIDs, id); !found {
+		pos, found := slices.BinarySearch(usedIDs, id)
+		if !found && !isReservedID(id) {
```

Comment:
But can't we just continue if reserved? Do we still need to check the list?

Reply:
yes, we can indeed optimize a bit more. I pushed a commit which does that.